AK/Tests/TestRefPtr.cpp

```diff
@@ -79,7 +79,7 @@ TEST_CASE(assign_copy_self)
 {
     RefPtr<Object> object = adopt(*new Object);
     EXPECT_EQ(object->ref_count(), 1);
-    object = object;
+    object = static_cast<RefPtr<Object>>(object);
```

Comment:
What was the warning here?

Reply:
I'd get `-Wself-assign-overloaded` here.

Reply:
Can we temporarily disable the warning instead?
Something like:
```cpp
#ifdef __clang__
#    pragma clang diagnostic push
#    pragma clang diagnostic ignored "-Wself-assign-overloaded"
#endif
    object = object;
#ifdef __clang__
#    pragma clang diagnostic pop
#endif